azure-iot-hub-devicesdk/samples/send_to_output_module.py

```diff
@@ -0,0 +1,50 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import os
+import time
+from azure.iot.hub.devicesdk.module_client import ModuleClient
+from azure.iot.hub.devicesdk.auth.authentication_provider_factory import from_connection_string
+from azure.iot.hub.devicesdk.message import Message
+import uuid
+
+# The connection string for a device should never be stored in code. For the sake of simplicity we're using an environment variable here.
+conn_str = os.getenv("IOTHUB_MODULE_CONNECTION_STRING")
+# The "Authentication Provider" is the object in charge of creating authentication "tokens" for the device client.
+auth_provider = from_connection_string(conn_str)
```

Comment:
for modules, from_environment() is the factory method that you should use.  No connection string is necessary.

Reply:
I actually thought that this is one of the options. When I see in node I see that module client has been constructed with other type of authentication providers specially in the sample it has been created with `ModuleClient.fromConnectionString(connectionString, Protocol);`

---
In reply to: [246254580](https://github.com/Azure/azure-iot-sdk-python-preview/pull/35#discussion_r246254580) [](ancestors = 246254580)

Reply:
technically both exist and would be correct but if we assume that our primary target is edge developers, modules are likely going to be used in the context of edge and therefore with from_environment(). thanks for making that change :)

---
In reply to: [246598483](https://github.com/Azure/azure-iot-sdk-python-preview/pull/35#discussion_r246598483) [](ancestors = 246598483,246254580)